File: src/main/java/pe/nanamochi/banchus/services/gameplay/performance/CalculatorType.java
package pe.nanamochi.banchus.services.gameplay.performance;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum CalculatorType {
  OSU_NATIVE("osu-native"),
  ROSU("rosu-pp");

  private final String alias;

  public static CalculatorType fromAlias(String alias) {
    for (CalculatorType type : values()) {
      if (type.alias.equalsIgnoreCase(alias)) {
        return type;
      }
    }
    throw new IllegalArgumentException("Unknown calculator type alias: " + alias);
  }
}

File: banchus-app/src/main/kotlin/pe/nanamochi/banchus/infrastructure/performance/CalculatorType.kt
package pe.nanamochi.banchus.infrastructure.performance

enum class CalculatorType(val alias: String) {
    OSU_NATIVE("osu-native"),
    ROSU("rosu-pp");

    companion object {
        fun fromAlias(alias: String?): CalculatorType =
            entries.find { it.alias.equals(alias, ignoreCase = true) } ?: ROSU
    }
}
